content/300-guides/300-upgrade-guides/200-upgrading-versions/030-upgrading-to-prisma-4.mdx

```diff
@@ -0,0 +1,418 @@
+
+### Rounding errors on big numbers in SQLite
+
+SQLite is a loosely-typed database. If your schema has a field with type `Int`, then Prisma prevents you from inserting a value larger than an integer. However, nothing prevents the database from directly accepting a bigger number. These manually-inserted big numbers cause rounding errors when queried.
+
+To avoid this problem, Prisma version 4.0.0 and later checks numbers on the way out of the database to verify that they fit within the boundaries of an integer. If a number does not fit, then Prisma throws a P2023 error, such as:
+
+```
+Inconsistent column data: Conversion failed:
+Value 9223372036854775807 does not fit in an INT column,
+try migrating the 'int' column type to BIGINT
+```
+
+#### Upgrade path
+
+If you use Prisma in conjunction with SQLite, then you need to find any code that queries `Int` fields and ensure that it handles any P2023 errors that might be returned.
+
+### Prisma no longer exports `Prisma.dmmf.schema` into the generated Prisma Client
+
+From version 4.0.0, Prisma no longer exports `Prisma.dmmf.schema` into the generated Prisma Client. This makes the generated Prisma Client much more efficient, and also avoids some memory leaks if you use Jest to carry out tests.
+
+Note:
+
+- This change does not affect the DMMF that Prisma passes to the generators.
+- You can use `getDmmf()`from `@prisma/internals` to access the schema property.
+- We still export `Prisma.dmmf.datamodel` into the generated Prisma Client.
+
+### Index configuration
+
+In Prisma 4, the `extendedIndexes` Preview feature will now become generally available. This includes the following index configuration options:
+
+- Length configuration of indexes, unique constraints and primary key constraints for MySQL (in Preview in versions 3.5.0 and later)
+- Sort order configuration of indexes, unique constraints and primary key constraints (in Preview in versions 3.5.0 and later)
+- New index types for PostgreSQL: Hash (in Preview in versions 3.6.0 and later) and GIN, GiST, SP-GiST and BRIN (in Preview in versions 3.14.0 and later)
+- Index clustering for SQL Server (in Preview in versions 3.13.0 and later)
+
+See our documentation on [Index configuration](/concepts/components/prisma-schema/indexes#index-configuration) for more details of these features.
+
+#### Upgrade path
+
+These can all be breaking changes if you were previously configuring these properties at the database level. In this case, you will need to:
+
+- upgrade to the new Prisma 4 packages following [these instructions](#upgrade-the-prisma-and-prismaclient-packages-to-prisma-4)
+- run `prisma db pull` afterwards to retrieve any existing configuration of indexes and constraints. This needs to be done before running any `prisma db push` or `prisma migrate dev` command, or you may lose any configuration that was defined in the database but not previously represented in the Prisma schema.
```

Comment:
```suggestion
1. upgrade to the new Prisma 4 packages following [these instructions](#upgrade-the-prisma-and-prismaclient-packages-to-prisma-4)
2. run `prisma db pull` afterwards to retrieve any existing configuration of indexes and constraints. This needs to be done before running any `prisma db push` or `prisma migrate dev` command, or you may lose any configuration that was defined in the database but not previously represented in the Prisma schema.
```

Reply:
There's a couple other places where I felt like the list had steps, so they should be numbered.

Reply:
Thanks, changed to numbers